### DISCUSSION TITLE
'Slim' docker container

### DISCUSSION THREAD

**Discussion by mike-reynolds-personal** [postgresml/postgresml#1036] [Q&A]
Are there any plans to make a slimmer docker container, as it currently runs at ~15Gb?

Not being a Python person I love the idea of having vector embedding support directly within the database, but 15Gb is prohibitive for deploying on Kubernetes and I cannot tell if or what can be cleaned out of the image post building.

Many thanks.

**Comment by montanalow** [comment]
The NVIDIA CUDA drivers are several GB each, and we support the largest set of hardware possible with the default docker image. My recollection is that you should be able to get down to a few GB docker image for a single target, although I have not validated that recently.

Have you tried building the Dockerfile here: https://github.com/postgresml/postgresml/blob/master/docker/Dockerfile

It may be worth building your image on base ubuntu instead of nvidia/cuda, to see how much you can reduce bloat. This is possible, but somewhat laborious since NVIDIA does not open source their drivers.

**Comment by mike-reynolds-personal** [comment]
Thanks for the response.
Yes I did try the build myself and it comes out the same.
The Nvidia driver size rings a bell, I'll try the base Ubuntu as suggested although, after testing yesterday, it's going to be a serious hit on performance... :(

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by montanalow** [reply]
If you dig through the git history you can probably find our old docker file that installed the Nvidia drivers in the base Ubuntu image, so that you keep GPU acceleration.
